worlds/mlss/__init__.py

```diff
@@ -0,0 +1,173 @@
+            language='English',
+            file_name='setup_en.md',
+            link='setup/en',
+            authors=['jamesbrq']
+        )
+    ]
+
+
+class MLSSSettings(settings.Group):
+    class RomFile(settings.UserFilePath):
+        """File name of the MLSS US rom"""
+        copy_to = "Mario & Luigi - Superstar Saga (U).gba"
+        description = "MLSS ROM File"
+        md5s = ["4b1a5897d89d9e74ec7f630eefdfd435"]
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+    rom_start: bool = True
+
+
+class MLSSWorld(World):
+    """
+    Adventure with Mario and Luigi together in the Beanbean Kingdom
+    to stop the evil cackletta and retrieve the Beanstar.
+    """
+    game = "Mario & Luigi Superstar Saga"
+    web = MLSSWebWorld()
+    data_version = 1
+    options_dataclass = MLSSOptions
+    options: MLSSOptions
+    settings: typing.ClassVar[MLSSSettings]
+    item_name_to_id = {name: data.code for name, data in item_table.items()}
+    location_name_to_id = {loc_data.name: loc_data.id for loc_data in all_locations}
+    required_client_version = (0, 4, 5)
+
+    excluded_locations = []
+
+    def generate_early(self) -> None:
+        self.excluded_locations = []
+        if self.options.chuckle_beans == 0:
+            self.excluded_locations += [location.name for location in all_locations if "Digspot" in location.name]
+        if self.options.castle_skip:
+            self.excluded_locations += [location.name for location in all_locations if "Bowser" in location.name]
+        if self.options.chuckle_beans == 1:
+            self.excluded_locations = [location.name for location in all_locations if location.id in hidden]
+        if self.options.skip_minecart:
+            self.excluded_locations += [LocationName.HoohooMountainBaseMinecartCaveDigspot]
+        if self.options.disable_surf:
+            self.excluded_locations += [LocationName.SurfMinigame]
+        if self.options.harhalls_pants:
+            self.excluded_locations += [LocationName.HarhallsPants]
+        if not self.options.coins:
+            self.excluded_locations += [location.name for location in all_locations if location in coins]
+
+    def create_regions(self) -> None:
+        create_regions(self, self.excluded_locations)
+        connect_regions(self)
+
+    def fill_slot_data(self) -> dict:
+        return {
+            "CastleSkip": self.options.castle_skip.value,
+            "SkipMinecart": self.options.skip_minecart.value,
+            "DisableSurf": self.options.disable_surf.value,
+            "HarhallsPants": self.options.harhalls_pants.value,
+            "ChuckleBeans": self.options.chuckle_beans.value,
+            "DifficultLogic": self.options.difficult_logic.value,
+            "Coins": self.options.coins.value
+        }
+
+    def generate_basic(self) -> None:
+        item = self.create_item("Mushroom")
+        self.multiworld.get_location(LocationName.ShopStartingFlag1, self.player).place_locked_item(item)
```

Comment:
Nowadays you can do:
`self.get_location(LocationName.ShopStartingFlag1r).place_locked_item(item)`

it's quite nice

*Low Priority*